[docs][notifications] update reference docs (#33065)
expo/expo: docs/pages/versions/unversioned/sdk/notifications.mdx

`expo-notifications` provides an API to fetch push notification tokens and to present, schedule, receive and respond to notifications.

This page is a reference documentation for the package. There are also comprehensive [guides available](/push-notifications/overview/).

> > **warning** Push notifications (remote notifications) functionality provided by `expo-notifications` is unavailable in Expo Go from SDK 53. A [development build](/develop/development-builds/introduction/) is required to use push notifications. Local notifications (in-app notifications) remain available in Expo Go.

Reviewer: ```suggestion
> **warning** Push notifications (remote notifications) functionality provided by `expo-notifications` will be unavailable in Expo Go for Android beginning in SDK 53. A [development build](/develop/development-builds/introduction/) will be required to use push notifications. Local notifications (in-app notifications) will remain available in Expo Go.
```

`expo-notifications` provides an API to fetch push notification tokens and to present, schedule, receive and respond to notifications.

> This page is a reference document for the package. There are also comprehensive [guides available](/push-notifications/overview/).

Reviewer 2: Let's use BoxLink here to redirect towards Push notification section. We don't have to specify that this is an API page since that is self-explanatory.

Author: done

}
```

> > The iOS APNs entitlement is _always_ set to 'development'. Xcode automatically changes this to 'production' during the archive.

Reviewer: ```suggestion
> The iOS APNs entitlement is _always_ set to 'development'. Xcode automatically changes this to 'production' in the archive generated by a release build.
```

## Notification events listeners

Notification events include incoming notifications, interactions your users perform with notifications (this can be tapping on a notification, or interacting with it via [notification categories](#manage-notification-categories-interactive-notifications)), and rare occasions when your notifications may be dropped.

> Several different listeners are exposed, and documented in [Push Notification Behaviors](/push-notifications/what-you-need-to-know/#push-notification-behaviors) section.

Reviewer 2: ```suggestion
Several listeners are exposed and documented in the [Push notification behaviors](/push-notifications/what-you-need-to-know/#push-notification-behaviors) section.
```

## Headless (Background) notifications

See the [definition](/push-notifications/what-you-need-to-know/#headless-background-notifications) of Headless Background Notifications in the [What you need to know](/push-notifications/what-you-need-to-know) guide.

To handle notifications while the app is in the background or not running, you need to do the following:

- Add `expo-task-manager` package to your project.
- [Configure background notifications](#background-notification-configuration).
- In your application code, set up a [background task](#registertaskasynctaskname) to run when the notification is received.

Then send a push notification which:

- Contains only the `data` key (no `title`, `body`)

> - Has `_contentAvailable: true` set for iOS - see the [Expo push notification service payload format](/push-notifications/sending-notifications/#message-request-format)

Reviewer 2: ```suggestion
- Has `_contentAvailable: true` set for iOS &mdash; see the [Expo push notification service payload format](/push-notifications/sending-notifications/#message-request-format)
```
